sdk/core/azure-core/azure/core/serialization.py

```diff
@@ -82,7 +82,19 @@ def _timedelta_as_isostr(value):
 
     return "P" + date + time
 
-
+def _timestr_as_isostr(o):
```

Comment:
nit: it's hard to come up with a good name for this helper, since we're checking for a `datetime`, `date`, or `time`... but maybe `_time_as_isostr` would be a more fitting name, since the object we're passing in is a non-string ("timestr" implies to me that we're being given a time as a string). What do you think?

Reply:
Also, since this has been moved out of the `AzureJSONEncoder.default` method, it might be nice to rename the parameter `o` to something more descriptive 🙂 Maybe `value` to align with `_timedelta_as_isostr`?